include/boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp

```diff
@@ -100,23 +100,40 @@ class range_segment_iterator
             Reference
         >
 {
+private:
+    static bool has_less_than_two_elements(Range const& r)
+    {
+        iterator_type first = range_iterator_begin<Range>::apply(r);
+        iterator_type last = range_iterator_end<Range>::apply(r);
+
+        return first == last || ++first == last;
+    }
+
 public:
     typedef typename range_iterator_type<Range>::type iterator_type;
 
     // default constructor
     range_segment_iterator()
-        : m_it()
+        : m_it(), m_has_less_than_two_elements()
```

Comment:
@mkaravel Thanks!

@barendgehrels Indeed scalar types should be zero-initialized assuming the compiler is standard-conformant. However Geometry should work also with non-scalar coordinate types and we shouldn't assume that a default ctor should initialize an object to an equivalent of 0. Is there a reason why do you think that value-initialization is better than explicit initialization with 0? This is one more, unnecessary requirement for coordinate type. The library already requires that it should be possible to construct an object with some int or double value, e.g. in distance_cross_track:

    return_type const quarter(0.25);

which also has its drawbacks but it's another story. Why not do this also for 0 and leave the default ctor out of this?